specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,568 @@
+If the contents of the files differ, the VOA library will calculate a merged view of both sources, consolidating all available information.
+In this example, the OpenPGP certificate in `/usr/share/voa/arch/package/default/openpgp/bbe960a25ea311d21d40669e93df2003ba9b90a2.openpgp`, while the OpenPGP certificate in `/etc/voa/arch/package/default/openpgp/bbe960a25ea311d21d40669e93df2003ba9b90a2.openpgp` is not.
+Due to [merging] semantics, the OpenPGP certificate is considered revoked.
+
+The VOA library has effectively found one _trust anchor_ verifier file, and one valid _artifact verifier_ file.
+This means that package files will be verified using the one _artifact verifier_ (`/usr/share/voa/arch:::cashier-system:1.0.0/package/default/openpgp/62cdb7020ff920e5aa642c3d4066950dd1f01f4d.openpgp`), which in turn is checked for validity based on the _trust anchor_ verifier (`/usr/share/voa/arch:::cashier-system:1.0.0/trust-anchor-package/default/openpgp/0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33.openpgp`).
+
+### Download tool verifies installation medium, without trust anchor
+
+In this example, we'll consider the use case of a download tool that uses VOA to verify the signature of a Fedora 41 installation medium.
+
+The download tool issues a lookup for OpenPGP verifiers from VOA, based on a single [os] string: `fedora:41`.
+The tool is configured to perform validation with _artifact verifiers_ (i.e. it specifies that it does not require the use of _trust anchors_).
+
+A VOA library will search in the [load path] `/usr/share/voa/` - in our example the only location with verifier data - and consider the basic verifier paths.
+In our example, it will find the following verifier file for the `fedora:41` _os_ string:
+
+```
+/usr/share/voa/fedora:41/image/installation-medium/openpgp/62cdb7020ff920e5aa642c3d4066950dd1f01f4d.openpgp
+```
+
+The VOA library will also check for verifier files in the other layers, but in in this example, there are none.
+
+The VOA library has found one _artifact verifier_ file (`/usr/share/voa/fedora:41/image/installation-medium/openpgp/62cdb7020ff920e5aa642c3d4066950dd1f01f4d.openpgp`) and proceeds to use it to verify the installation medium file.
+
+## Considerations for implementers and users
+
+### Avoiding duplication of data in the directory structure
+
+Symlinks are explicitly supported when forming VOA structures.
+In many cases, it will be desirable to make heavy use of symlinks, to form the appropriate directory structures while avoiding duplication of data.
+However, the [VOA hierarchy] is intended as a self-contained data structure that specifies how to authenticate signatures.
+Therefore, symlinks in VOA should only point to other files or directories in any of the [VOA hierarchy] paths.
+Symlinks may be relative or absolute.
+Using files external to the VOA structure is not permitted and should lead to an error.
+
+The use of hardlinks in the VOA structure is not supported and should lead to an error.
```

Comment:
how is a hardlink supposed to lead to an error?  does the VOA implementation read `st_nlink` and yield an error if it is > 1?  what do we gain from this constraint?

Reply:
What this sentence was meant to convey is that hardlinks are confusing and should not be used.

Not checking for this explicitly in a VOA implementation is fine though and this should not be a hard requirement for implementers.

We will adjust the text accordingly.